tests/test_data_model_cache.py

```diff
@@ -0,0 +1,198 @@
+    initial_cache, rules, state, deleted_paths, final_cache
+):
+    DataModelCache.rules_str_to_cache.clear()
+    DataModelCache.rules_str_to_cache.update(initial_cache)
+    var = Variant()
+    _convert_value_to_variant(state, var)
+    DataModelCache.update_cache(rules, var, deleted_paths)
+    assert DataModelCache.rules_str_to_cache == final_cache
+
+
+@pytest.mark.parametrize(
+    "initial_cache,rules,state,deleted_paths,final_cache",
+    [
+        (
+            {"r1": {"A": {"B": "ab", "C": {}, "D:D1": {}}}},
+            "r1",
+            None,
+            ["X/D:D1", "A/D:Y1", "A/D:D1"],
+            {"r1": {"A": {"B": "ab", "C": {}}}},
+        ),
+        ({"r1": {}}, "r1", {"A": True}, [], {"r1": {"A": True}}),
+        ({"r1": {}}, "r1", {"A": 5}, [], {"r1": {"A": 5}}),
+        ({"r1": {}}, "r1", {"A": 3.0}, [], {"r1": {"A": 3.0}}),
+        ({"r1": {}}, "r1", {"A": "ab"}, [], {"r1": {"A": "ab"}}),
+        ({"r1": {}}, "r1", {"A": [False, True]}, [], {"r1": {"A": [False, True]}}),
+        ({"r1": {}}, "r1", {"A": [5, 10]}, [], {"r1": {"A": [5, 10]}}),
+        ({"r1": {}}, "r1", {"A": [3.0, 6.0]}, [], {"r1": {"A": [3.0, 6.0]}}),
+        ({"r1": {}}, "r1", {"A": ["ab", "cd"]}, [], {"r1": {"A": ["ab", "cd"]}}),
+        ({"r1": {"A": {}}}, "r1", {"A": {"B": 5}}, [], {"r1": {"A": {"B": 5}}}),
+        (
+            {"r1": {"A": {}}},
+            "r1",
+            {"A": {"B": {"C": 5}}},
+            [],
+            {"r1": {"A": {"B": {"C": 5}}}},
+        ),
+        (
+            {"r1": {"A": {}}},
+            "r1",
+            {"A": {"B:B1": {"_name_": "B-1", "C": 5.0}}},
+            [],
+            {"r1": {"A": {"B:B1": {"_name_": "B-1", "C": 5.0}}}},
+        ),
+        (
+            {"r1": {"A": {"B:B1": {"_name_": "B-1", "C": 5.0}}}},
+            "r1",
+            {"A": {"B:B1": {"C": 7.0}}},
+            [],
+            {"r1": {"A": {"B:B1": {"_name_": "B-1", "C": 7.0}}}},
+        ),
+        (
+            {"r1": {}},
+            "r1",
+            {"B:B1": {"_name_": "B-1", "C": 5.0}},
+            [],
+            {"r1": {"B:B1": {"_name_": "B-1", "C": 5.0}}},
+        ),
+        (
+            {"r1": {"B:B1": {"_name_": "B-1", "C": 5.0}}},
+            "r1",
+            {"B:B1": {"C": 7.0}},
+            [],
+            {"r1": {"B:B1": {"_name_": "B-1", "C": 7.0}}},
+        ),
+        (
+            {"r1": {}},
+            "r1",
+            {"A:A1": {"_name_": "A-1", "B:B1": {"_name_": "B-1", "D": 7.0}, "C": 5.0}},
+            [],
+            {
+                "r1": {
+                    "A:A1": {
+                        "_name_": "A-1",
+                        "B:B1": {"_name_": "B-1", "D": 7.0},
+                        "C": 5.0,
+                    }
+                }
+            },
+        ),
+        (
+            {"r1": {"B:B1": {"_name_": "B-1", "C": 5.0}}},
+            "r1",
+            {"B:B1": {"_name_": "B-2", "C": 5.0}},
+            ["B:B1"],
```

Comment:
@seanpearsonuk Checked that this is streamed on renaming a named object in Fluent - deletion + creation